### ISSUE TITLE
Support display names for fuzzers and jobs

### ISSUE THREAD

**Issue by oliverchang** [google/clusterfuzz#796]
Generalisation of #781 for discussion.

I think it would be nice to be able to specify a pretty format for fuzzers and jobs. 

For example, a fully qualified name of "libFuzzer_a-b-c-d-e-fuzzer" might have a display name set to something like: "libFuzzer: //a/b/c/d/e:fuzzer" Similar for jobs.

We can display these everywhere in our UI rather than displaying potentially ugly and difficult to read strings.

**Comment by Dor1s** [comment]
What is `fuzzer` in this example, a binary name? If so, do we essentially need to customize the format only for `a-b-c-d-e-` or (kinda) project name?

**Comment by oliverchang** [comment]
Yep, `fuzzer` would be the binary name. I think it may be nice to use this format everywhere, since our job and fully qualified names may not be immediately obvious.

**Comment by Dor1s** [comment]
Do we want to control it on CF side or from within the build archive?
